.github/workflows/ci.yml

```diff
@@ -3,7 +3,41 @@ on:
   push:
     branches: [master]
   pull_request:
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
 jobs:
+  install_dependencies:
```

Comment:
this mostly primes the cache, which the other tasks than utilize

Reply:
We can keep an eye on this going forward. It seems like, at least for now, either way is fine.